onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -121,18 +118,41 @@ static bool IsRelevantOutput(const Node* node, const NodeArg* output) {
 
   auto it = opcode_to_output_map.find(node->OpType());
   if (it != opcode_to_output_map.cend()) {
-    const std::vector<int>& selected_outputs = it->second;
+    const auto& selected_outputs = it->second;
     int input_index = optimizer_utils::IndexOfNodeOutput(*node, *output);
     return std::find(selected_outputs.begin(), selected_outputs.end(), input_index) != selected_outputs.end();
   }
   return true;
 }
 
+namespace {
+// borrowed from providers/common.h
```

Comment:
Why is this change (substitution) required?

Reply:
This is a readability change based on the similar providers function. The pattern is easily recognizable as it is used in other parts of the code.

Reply:
Is the existing pattern not used any other parts of the code?

Reply:
some absl containers have a contains() method